packages/react-native-editor/__device-tests__/pages/editor-page.js

```diff
@@ -261,18 +285,40 @@ class EditorPage {
 	}
 
 	async dismissKeyboard() {
+		const orientation = await this.driver.getOrientation();
 		const keyboardShown = await this.driver.isKeyboardShown();
 		if ( ! keyboardShown ) {
 			return;
 		}
-		if ( isAndroid() ) {
+
+		// On Android with the landspace orientation set, we use the
+		// driver functionality to hide the keyboard.
+		if ( isAndroid() && orientation === 'LANDSCAPE' ) {
 			return await this.driver.hideDeviceKeyboard();
 		}
 
-		await clickIfClickable(
-			this.driver,
-			'//XCUIElementTypeButton[@name="Hide keyboard"]'
-		);
+		const hideKeyboardButton = isAndroid()
+			? await this.waitForElementToBeDisplayedById(
+					'Hide keyboard, Tap to hide the keyboard'
+			  )
+			: await this.waitForElementToBeDisplayedByXPath(
+					'//XCUIElementTypeButton[@name="Hide keyboard"]'
+			  );
+
+		await hideKeyboardButton.click();
+		await this.waitForKeyboardToBeHidden();
+	}
+
+	// Takes the add block button as reference for the keyboard to be
+	// fully hidden.
+	async waitForKeyboardToBeHidden() {
+		const { addButtonLocation } = this.initialValues;
+		const addButton = await this.getAddBlockButton();
+		const location = await addButton.getLocation();
+
+		if ( location.y < addButtonLocation?.y ) {
+			await this.waitForKeyboardToBeHidden();
+		}
 	}
```

Comment:
I noticed that Appium provides a way to check if the keyboard is shown ([`isKeyboardShown`](https://appium.io/docs/en/commands/device/keys/is-keyboard-shown/)). In fact, we [use it on this function](https://github.com/WordPress/gutenberg/blob/298f7d1bc69d6fc5e1b2efdc1fff7386b2c44585/packages/react-native-editor/__device-tests__/pages/editor-page.js#L289), so I'm wondering if we could use it too for waiting for the keyboard to be hidden, WDYT?

Reply:
I thought about it but the issue is that the component where the toolbar, and Add button are, takes some time to go back to its position final position so it causes issues when matching screenshots where sometimes you'll see the keyboard hidden but the bottom bar of the editor hasn't reached the edge yet 😅 

I'd like to try this and see if it's stable enough in CI 🤞